libcontainer/integration/template_test.go

```diff
@@ -191,4 +199,28 @@ func newTemplateConfig(rootfs string) *configs.Config {
 			},
 		},
 	}
+
+	if p.userns {
+		config.UidMappings = []configs.IDMap{{HostID: 0, ContainerID: 0, Size: 1000}}
+		config.GidMappings = []configs.IDMap{{HostID: 0, ContainerID: 0, Size: 1000}}
+		config.Namespaces = append(config.Namespaces, configs.Namespace{Type: configs.NEWUSER})
+	} else {
```

Comment:
This function will be a long, long list of `ifs` if more options are added; wondering, would it be an idea to use functional arguments for this? `newTemplateConfig(withRootfs(foo), withSystemd())`?

Reply:
Frankly speaking I just don't like that paradigm much. I find it harder to read as you have to jump back and forth too much to read the code. It might be useful for APIs with external users, to make the code using it less ugly, but in this case these are internal tests, so the only user is runc.

Actually, let me try rewriting it with functional args, and let's see which one looks better...

Reply:
OK, I tried, but I am not sure how to handle "else" cases. Such as this one:

```go
        if p.userns {
                config.UidMappings = []configs.IDMap{{HostID: 0, ContainerID: 0, Size: 1000}}
                config.GidMappings = []configs.IDMap{{HostID: 0, ContainerID: 0, Size: 1000}}
                config.Namespaces = append(config.Namespaces, configs.Namespace{Type: configs.NEWUSER})
        } else {
                config.Mounts = append(config.Mounts, &configs.Mount{
                        Destination: "/sys/fs/cgroup",
                        Device:      "cgroup",
                        Flags:       defaultMountFlags | unix.MS_RDONLY,
                })
        }
```

Surely I can add this mount unconditionally, and make `withUserNS` remove it, but it does not look perfect :-\

So, I'd rather let it stay as is for now.

Reply:
Feel free to code the alternative though if you like @thaJeztah